src/java/org/apache/cassandra/index/sai/disk/v2/V2VectorIndexSearcher.java

```diff
@@ -214,9 +237,9 @@ private BitsOrPostingList bitsOrPostingListForKeyRange(QueryContext context,
             cost = estimateCost(topK, bits.cardinality());
 
             if (!hasMatches)
-                return new BitsOrPostingList(PostingList.EMPTY);
+                return BitsOrRowIds.EMPTY;
```

Comment:
You're right, I was able to get rid of the `BitsOrPostingList` class entirely. There was a slight nuance in the way we handle cost estimates, so I ended up nesting the logic within this method instead of returning row ids and then deciding whether to search the graph.